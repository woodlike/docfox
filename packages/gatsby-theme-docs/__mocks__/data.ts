export const dataFrontMatterComplete = `
---
name: Video
menu: Components
title: Video
---

import { Video } from '@wdlk/components';
import { useVideoControl } from '@wdlk/hooks';
import { useState } from 'react';

# Video Stage

Art directing videos on three different devices.

## Fully functional Video Stage Example
`;

export const resultDataComplete = `
import { Video } from '@wdlk/components';
import { useVideoControl } from '@wdlk/hooks';
import { useState } from 'react';

# Video Stage

Art directing videos on three different devices.

## Fully functional Video Stage Example
`;

export const dataFrontMatterNameOnly = `
---
name: Video33
---

import { Video } from '@wdlk/components';
import { useVideoControl } from '@wdlk/hooks';
import { useState } from 'react';

# Video Stage

Art directing videos on three different devices.

## Fully functional Video Stage Example
`;

export const resultNameOnly = `
import { Video } from '@wdlk/components';
import { useVideoControl } from '@wdlk/hooks';
import { useState } from 'react';

# Video Stage

Art directing videos on three different devices.

## Fully functional Video Stage Example
`;

export const dataFrontMatterVideoName = `
---
name: Code
menu: Components
---

# Code

## The Prism highlighter component for React.

Introduction
`;

export const resultVideoName = `
# Code

## The Prism highlighter component for React.

Introduction
`;

export const dataNoFrontMatter = `
import { Heading } from '@wdlk/components';

## Heading Families

Covering three families as defined in the theme heading props.

- **display**,
- **secondary**
- **campaign**

### Examples

<Heading as="h2" size="xxl">
  Healing Flowers Collection
</Heading>
<Heading as="h2" size="xxl" family="secondary">
  Healing Flowers Collection
</Heading>
<Heading as="h2" size="xxl" family="campaign">
  Healing Flowers Collection
</Heading>
`;
