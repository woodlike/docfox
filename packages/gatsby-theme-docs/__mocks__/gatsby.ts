export const gatsby = {
  ...jest.requireActual('gatsby'),
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
  reporter: {
    panicOnBuild: jest.fn().mockName('panicOnBuild'),
  },
};
