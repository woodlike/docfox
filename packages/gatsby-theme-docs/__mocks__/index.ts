export * from './data';
export * from './gatsby';
export * from './mdxAstMock';
