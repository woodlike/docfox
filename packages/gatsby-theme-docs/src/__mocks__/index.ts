export * from './data';
export * from './docs';
export * from './gatsby';
export * from './mdxAstMock';
