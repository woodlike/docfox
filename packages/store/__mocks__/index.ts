export * from './navigation-data-mock';
export * from './header-data-mock';
