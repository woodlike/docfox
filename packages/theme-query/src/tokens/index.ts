export * from './borders';
export * from './colors';
export * from './fonts';
export * from './spaces';
export * from './transitions';
