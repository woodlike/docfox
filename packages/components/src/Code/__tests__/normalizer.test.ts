import { Token } from 'prismjs';
import { normalizer } from '..';
import { prismTokens } from '../__mocks__/tokens';

describe('normalizer()', () => {
  describe('Error handling', () => {
    beforeEach(() => {
      global.console = ({ error: jest.fn() } as unknown) as Console;
    });

    it('should log an error on missing styles prop', () => {
      const tokens = normalizer({} as (string | Token)[]);
      expect(tokens).toEqual([]);
      expect(console.error).toHaveBeenCalled();
      expect(console.error).toHaveBeenCalledWith(
        'Error: The provided token stream must be of type array returned by the Prism.tokenize(code, grammar)',
      );
    });
  });

  describe('Flattened Token Array', () => {
    let prismTokensMock: (string | Token)[];
    beforeEach(() => {
      prismTokensMock = [
        {
          type: 'tag',
          content: [
            {
              type: 'tag',
              content: [
                {
                  type: 'punctuation',
                  content: '<',
                },
                'pre',
              ],
            },
            ' ',
            {
              type: 'attr-name',
              content: ['sx'],
            },
            {
              type: 'script',
              content: [
                {
                  type: 'script-punctuation',
                  content: '=',
                },
                {
                  type: 'punctuation',
                  content: '{',
                },
                {
                  type: 'function',
                  content: 'createStylesPre',
                },
                'props',
              ],
              alias: 'language-javascript',
            },
          ],
        },
        '\n    ',
      ] as (string | Token)[];
    });

    afterEach(() => {
      prismTokensMock = (undefined as unknown) as (string | Token)[];
    });

    it('should return a flattened token array from the prismTokensMock Mock', () => {
      const result = [
        { type: 'punctuation', content: '<' },
        { type: 'tag', content: 'pre', alias: '', length: 0, greedy: false },
        { type: 'tag', content: ' ', alias: '', length: 0, greedy: false },
        { type: 'attr-name', content: 'sx', alias: '', length: 0, greedy: false },
        { type: 'script-punctuation', content: '=' },
        { type: 'punctuation', content: '{' },
        { type: 'function', content: 'createStylesPre' },
        { type: 'script', content: 'props', alias: '', length: 0, greedy: false },
        { type: 'tag', content: '\n    ', alias: '', length: 0, greedy: false },
      ];
      expect(normalizer(prismTokensMock)).toEqual(result);
    });
    it('should return a flattened token array from a returned by Prism.tokenize() (___mocks___)', () => {
      expect(normalizer(prismTokens)).toMatchSnapshot();
    });
  });
});
