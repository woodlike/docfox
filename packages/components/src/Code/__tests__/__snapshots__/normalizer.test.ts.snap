// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizer() Flattened Token Array should return a flattened token array from a returned by Prism.tokenize() (___mocks___) 1`] = `
Array [
  Object {
    "content": "<",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "pre",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "alias": "",
    "content": "sx",
    "greedy": false,
    "length": 0,
    "type": "attr-name",
  },
  Object {
    "alias": "punctuation",
    "content": "=",
    "greedy": false,
    "length": 1,
    "type": "script-punctuation",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "createStylesPre",
    "greedy": false,
    "length": 15,
    "type": "function",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "props",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "size",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ",",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " qt",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ",",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " props",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "theme",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
    ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "<",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "code",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "alias": "",
    "content": "sx",
    "greedy": false,
    "length": 0,
    "type": "attr-name",
  },
  Object {
    "alias": "punctuation",
    "content": "=",
    "greedy": false,
    "length": 1,
    "type": "script-punctuation",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "stylesCode",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
      ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "tokens",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "map",
    "greedy": false,
    "length": 3,
    "type": "function",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "token",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ":",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " Token ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "|",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " string",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ",",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " i",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ":",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " number",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": ":",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "JSX",
    "greedy": false,
    "length": 3,
    "type": "constant",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "Element",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "=>",
    "greedy": false,
    "length": 2,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
        console",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "log",
    "greedy": false,
    "length": 3,
    "type": "function",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "tokens",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": ";",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
        ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "return",
    "greedy": false,
    "length": 6,
    "type": "keyword",
  },
  Object {
    "alias": "",
    "content": " token",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "type ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "?",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
          Array",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "isArray",
    "greedy": false,
    "length": 7,
    "type": "function",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "token",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "content",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "?",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
            ",
    "greedy": false,
    "length": 0,
    "type": "parameter",
  },
  Object {
    "content": "<",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "TokenList",
    "greedy": false,
    "length": 9,
    "type": "class-name",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "alias": "",
    "content": "content",
    "greedy": false,
    "length": 0,
    "type": "attr-name",
  },
  Object {
    "alias": "punctuation",
    "content": "=",
    "greedy": false,
    "length": 1,
    "type": "script-punctuation",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "token",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "content",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "alias": "",
    "content": "theme",
    "greedy": false,
    "length": 0,
    "type": "attr-name",
  },
  Object {
    "alias": "punctuation",
    "content": "=",
    "greedy": false,
    "length": 1,
    "type": "script-punctuation",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "theme",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": "/>",
    "greedy": false,
    "length": 2,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
          ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ":",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
            ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "<",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "span",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "alias": "",
    "content": "sx",
    "greedy": false,
    "length": 0,
    "type": "attr-name",
  },
  Object {
    "alias": "punctuation",
    "content": "=",
    "greedy": false,
    "length": 1,
    "type": "script-punctuation",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "theme",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "get",
    "greedy": false,
    "length": 3,
    "type": "function",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "token",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "type",
    "greedy": false,
    "length": 0,
    "type": "script",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
              ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "typeof",
    "greedy": false,
    "length": 6,
    "type": "keyword",
  },
  Object {
    "alias": "",
    "content": " token",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "content ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "===",
    "greedy": false,
    "length": 3,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "'string'",
    "greedy": true,
    "length": 8,
    "type": "string",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "&&",
    "greedy": false,
    "length": 2,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " token",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ".",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "content",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
            ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "</",
    "greedy": false,
    "length": 2,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "span",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
          ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
        ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ":",
    "greedy": false,
    "length": 1,
    "type": "operator",
  },
  Object {
    "alias": "",
    "content": " ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "(",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
          ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "<",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "span",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "{",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "token",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "</",
    "greedy": false,
    "length": 2,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "span",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
        ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": ";",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
      ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": ")",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "content": "}",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
    ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "</",
    "greedy": false,
    "length": 2,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "code",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "
  ",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": "</",
    "greedy": false,
    "length": 2,
    "type": "punctuation",
  },
  Object {
    "alias": "",
    "content": "pre",
    "greedy": false,
    "length": 0,
    "type": "tag",
  },
  Object {
    "content": ">",
    "greedy": false,
    "length": 1,
    "type": "punctuation",
  },
]
`;
